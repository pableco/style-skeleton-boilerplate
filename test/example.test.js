/* eslint-env node, jest */
/* eslint "import/no-extraneous-dependencies": ["error", {"devDependencies": true }] */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-statements */

import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Test from '../src';
import ListItem from '../src/dynamic_templates/modules/list';

const props = {
    classname: '--news',
};

describe('component snapshoot', () => {

    it('render without props', () => {
        const wrapper = renderer.create(
            <Test />,
        ).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('render with all props', () => {
        const wrapper = renderer.create(
            <Test {...props}/>,
        ).toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});

describe('ListItem', () => {

    it('example', () => {
        const wrapper = mount(<ListItem />);

        expect(wrapper.find('li')).toHaveLength(1);
    });

});

describe('List test', () => {
    let data;

    beforeEach(() => {
        data = {
            ...props,
        };

    });

    it('renders a list', () => {
        const wrapper = shallow(<Test {...data} />);

        expect(wrapper.type()).toBe('ul');
    });

    it('check length of the list', () => {
        const wrapper = shallow(<Test {...data} />);

        expect(wrapper.find('ListItem')).toHaveLength(5);
    });

});
