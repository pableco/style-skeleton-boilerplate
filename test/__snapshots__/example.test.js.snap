// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshoot render with all props 1`] = `
<li
  className="list-item js--hover"
>
  <figure
    className="list-figure"
  >
    <figcaption
      className="list-figcaption"
    >
      <h3
        className="list-item__title"
      >
        Test title
      </h3>
      <h4
        className="list-item__sub-title"
      >
        Subtitle test
      </h4>
    </figcaption>
    <div
      className="list-item__img-fix"
      style={
        Object {
          "backgroundImage": "url(https://source.unsplash.com/random)",
        }
      }
    />
    <img
      alt="Test title image"
      className="list-item__img"
      src="https://source.unsplash.com/random"
    />
    <div
      className="list-item__content"
    >
      <p
        className="list-item__description"
      >
        Aenean vitae neque ultricies, egestas eros non, bibendum purus.
    Suspendisse sodales quam non lectus imperdiet dignissim.
      </p>
      <button
        className="list-item__action"
      >
        Call now
      </button>
    </div>
    <span
      className="list-item__span"
    />
  </figure>
</li>
`;

exports[`component snapshoot render without props 1`] = `
<li
  className="list-item js--hover"
>
  <figure
    className="list-figure"
  >
    <figcaption
      className="list-figcaption"
    >
      <h3
        className="list-item__title"
      >
        Test title
      </h3>
      <h4
        className="list-item__sub-title"
      >
        Subtitle test
      </h4>
    </figcaption>
    <div
      className="list-item__img-fix"
      style={
        Object {
          "backgroundImage": "url(https://source.unsplash.com/random)",
        }
      }
    />
    <img
      alt="Test title image"
      className="list-item__img"
      src="https://source.unsplash.com/random"
    />
    <div
      className="list-item__content"
    >
      <p
        className="list-item__description"
      >
        Aenean vitae neque ultricies, egestas eros non, bibendum purus.
    Suspendisse sodales quam non lectus imperdiet dignissim.
      </p>
      <button
        className="list-item__action"
      >
        Call now
      </button>
    </div>
    <span
      className="list-item__span"
    />
  </figure>
</li>
`;
